tests/serde.rs

```diff
@@ -1,44 +1,41 @@
 #![cfg(feature = "serde_impl")]
 
 extern crate linked_hash_map;
-extern crate serde;
-extern crate serde_json;
-
 use linked_hash_map::LinkedHashMap;
 
+extern crate serde_test;
+use serde_test::{Token, assert_tokens};
+
 #[test]
-fn test_ser_empty() {
+fn test_ser_de_empty() {
     let map = LinkedHashMap::<String, u32>::new();
-    let j = serde_json::to_string(&map).unwrap();
-    let expected = "{}";
-    assert_eq!(j, expected);
+
+    assert_tokens(&map, &[
+        Token::MapStart(Some(0)),
+        Token::MapEnd,
+    ]);
 }
 
 #[test]
-fn test_ser() {
+fn test_ser_de() {
     let mut map = LinkedHashMap::new();
-    map.insert("b", 20);
-    map.insert("a", 10);
-    map.insert("c", 30);
+    map.insert("b".to_string(), 20);
```

Comment:
It is possible to use &str for serializing but not for deserializing (because what would the lifetime be). This test is testing both so it was simpler to use String.